src/commands/transaction/secretlock.ts

```diff
@@ -0,0 +1,171 @@
+    })
+    url: string;
+
+    @option({
+        description: 'The network type of the other chain. (MAIN_NET, TEST_NET, MIJIN, MIJIN_TEST)',
+        flag: 'n',
+        validator: new NetworkValidator(),
+    })
+    networkType: string;
+}
+
+@command({
+    description: ' Creates a secretLock transaction ',
+})
+export default class extends AnnounceTransactionsCommand {
+    constructor() {
+        super();
+    }
+
+    @metadata
+    execute(options: CommandOptions) {
+        options.mosaicId = OptionsResolver(options,
+            'mosaicId',
+            () => undefined,
+            'Introduce the locked mosaic ID: ');
+
+        options.duration = OptionsResolver(options,
+            'duration',
+            () => undefined,
+            'Number of blocks for which a lock should be valid: ');
+
+        options.hashAlgorithm = OptionsResolver(options,
+            'hashAlgorithm',
+            () => undefined,
+            'Introduce algorithm used to hash the proof(0: Op_Sha3_256, 1: Op_Keccak_256, 2: Op_Hash_160, 3: Op_Hash_256): ');
+
+        options.secret = OptionsResolver(options,
+            'secret',
+            () => undefined,
+            'Introduce proof hashed: ');
+        options.recipientAddress = OptionsResolver(options,
+            'recipientAddress',
+            () => undefined,
+            'Introduce proof hashed: ');
+        options.privateKey = OptionsResolver(options,
+            'privateKey',
+            () => undefined,
+            'Introduce the private key of your account on the other chain: ');
+        options.url = OptionsResolver(options,
+            'url',
+            () => undefined,
+            'Introduce the url of the other chain: ');
+        options.networkType = OptionsResolver(options,
+            'networkType',
+            () => undefined,
+            'Introduce the network type of the other chain: (MAIN_NET, TEST_NET, MIJIN, MIJIN_TEST)');
+
+        const profile = this.getProfile(options);
+
+        const otherChainAccount = Account.createFromPrivateKey(options.privateKey, profile.networkType);
+
+        const secretLockService = new SecretLockService();
+        const {proof, secret} = secretLockService.createSecret();
+
+        console.log(chalk.green('proof: ') + proof + '\n');
+        console.log(chalk.green('secret: ') + secret + '\n');
+
+        const networkType = secretLockService.getNetworkType(options.networkType);
+        const cryptoType = secretLockService.getCryptoType(options.hashAlgorithm);
+
+        const secretLockTransaction = SecretLockTransaction.create(
+            Deadline.create(),
+            new Mosaic(new MosaicId(options.mosaicId),
+            UInt64.fromUint(options.amount)),
+            UInt64.fromUint(options.duration),
+            cryptoType,
+            secret,
+            Address.createFromRawAddress(options.recipientAddress),
+            networkType);
+
+        const blockHttp = new BlockHttp(options.url);
```

Comment:
It's not necessary to pass the other chain private key or url. Only define, sign and announce the secret lock transaction using the profile.url, as you are doing in the secret proof command.

Example: https://nemtech.github.io/guides/secretlock/atomic-cross-chain-swap-between-NEM-public-and-private-chain.html

Reply:
@NemTechCN

Reply:
[https://github.com/nemtech/nem2-docs/blob/master/source/resources/examples/typescript/secretlock/CrossChainSwap.ts](https://github.com/nemtech/nem2-docs/blob/master/source/resources/examples/typescript/secretlock/CrossChainSwap.ts)

I wrote my code according to this demo. In this demo, Alice signed the secret lock transaction with `alicePrivateChainAccount`. So I guess Alice should have an account in the private chain before she announce the secret lock transaction. And the same to `url`, `networkType` (in private chain).